sdk/src/main/java/io/radar/sdk/model/RadarUser.kt

```diff
@@ -143,9 +155,21 @@ class RadarUser(
             val dma = obj.optJSONObject(FIELD_DMA)?.let(RadarRegion.Companion::fromJson)
             val postalCode = obj.optJSONObject(FIELD_POSTAL_CODE)?.let(RadarRegion.Companion::fromJson)
 
-            val nearbyChains = obj.optJSONArray(FIELD_NEARBY_PLACE_CHAINS)?.let { nearbyArr ->
-                (0 until nearbyArr.length()).map { idx ->
-                    RadarChain.fromJson(nearbyArr.optJSONObject(idx) ?: JSONObject())
+            val nearbyChains = obj.optJSONArray(FIELD_NEARBY_PLACE_CHAINS)?.let { nearbyChainsArr ->
+                (0 until nearbyChainsArr.length()).map { idx ->
+                    RadarChain.fromJson(nearbyChainsArr.optJSONObject(idx) ?: JSONObject())
+                }
+            }?.toTypedArray()
+
+            val segments = obj.optJSONArray(FIELD_SEGMENTS)?.let { segmentsArr ->
```

Comment:
can we use the `fromJsonArray` in `RadarSegment` here?

Reply:
Good call, updating

Reply:
Done. I may do a bit more refactoring here before `beta.1`